test/test_set_date.py

```diff
@@ -1,26 +1,22 @@
-import unittest
+from datetime import datetime
+
+import pytest
 
 import fpdf
 from test.utilities import assert_pdf_equal
 
-from datetime import datetime
-
 
-class CreationDateTest(unittest.TestCase):
+class TestCreationDate:
     def test_setting_bad_date(self):
         doc = fpdf.FPDF()
         doc.set_creation_date("i am not a date")
-        with self.assertRaises(BaseException):
+        with pytest.raises(BaseException):
```

Comment:
Is there a reason to use `BaseException` instead of the standard `Exception`? 🤔

Reply:
ideally it would raise FPDFException

https://github.com/PyFPDF/fpdf2/blob/e318914ca5935b7e3e6dc350f97c1f95012a1fd2/fpdf/errors.py#L1

Reply:
There are several such cases. I have added an extra commit with them.